curator-recipes/src/test/java/org/apache/curator/framework/recipes/cache/TestTreeCache.java

```diff
@@ -426,10 +426,20 @@ public void testKilledSession() throws Exception
         assertEvent(TreeCacheEvent.Type.NODE_ADDED, "/test/me");
 
         KillSession2.kill(client.getZookeeperClient().getZooKeeper());
-        assertEvent(TreeCacheEvent.Type.CONNECTION_LOST);
-        assertEvent(TreeCacheEvent.Type.CONNECTION_RECONNECTED);
-        assertEvent(TreeCacheEvent.Type.INITIALIZED);
-        assertEvent(TreeCacheEvent.Type.NODE_REMOVED, "/test/me", "data".getBytes());
+        if ( client.isZk34CompatibilityMode() )
```

Comment:
Is this a race condition? I reverted this test and it seems to run fine for me (the 5 times I've run it)

Reply:
I'm going to re-work this test